source/Ship.cpp

```diff
@@ -3759,6 +3790,8 @@ void Ship::ExpendAmmo(const Weapon &weapon)
 		// A realistic fraction applicable to all cases cannot be computed, so assume 50%.
 		heat -= weapon.AmmoUsage() * .5 * ammo->Mass() * MAXIMUM_TEMPERATURE * Heat();
 		AddOutfit(ammo, -weapon.AmmoUsage());
+		if(!OutfitCount(ammo))
```

Comment:
Maybe check for `weapon.AmmoUsage()` here as well, so that the deterrence doesn't always get recalculated for weapons with ammo but no ammo usage.